### PR TITLE
fix: Addresses Samsung inconsistencies and fontScaling issues

### PR DESCRIPTION
**Pull request by juan-cortes**
For larger font sizes we had two remaining issues that were mainly present on Samsung devices. First of all, the footer modals that had bullet point lists were shown with huge texts that broke the design. Since this is the first approach at accessibility where we stop the font change altogether to have a stable base for future accessibility we needed to address that.

The issue was caused by[ nested text components](https://github.com/facebook/react-native/issues/19833), causing the `allowFontScaling` prop to be lost at the parent level and rendering the child elements, in this case, the text points of the list, with the scaled size.

The second error was affecting only Samsung devices with font scaling and it made password inputs unusable. This is caused by not using our wrapper for text input and relying on the native one that doesn't have the font scaling recalculation. That is also solved in this pr.

### Type

Fix

### Context

LL-1178

### Parts of the app affected / Test plan

Same as the other time we had a font scaling test plan, go over the app with scale mode enabled and see if there are more broken things. There is one known issue that I've been unable to fix so far, but it's so much more edge case than before that I think we can safely ignore it for now.

### Known issue
With font scaling enabled, on Android, on a text input that allows you to clear the input, the font size goes pretty small (half size) after clearing with the Right X button.

### CODE REVIEW

File: src/components/TextInput.ios.js [hunk 1]
```diff
@@ -22,6 +23,7 @@ class TextInput extends PureComponent<*> {
       <ReactNativeTextInput
         ref={innerRef}
         allowFontScaling={false}
+        style={[containerStyle, style]}
```
Comment (gre) [inline src/components/TextInput.ios.js]: this PR fixes Samsung thing, why is there a change needed on iOS impl too?

File: src/index.js [hunk 2]
```diff
@@ -38,9 +38,14 @@ const styles = StyleSheet.create({
 
 // Fixme until third parties address this themselves
 // $FlowFixMe
-Text.defaultProps = Text.defaultProps || {};
+Text.defaultProps = { ...(Text.defaultProps || {}), allowFontScaling: false };
```
Comment (gre) [inline src/index.js]:
```suggestion
Text.defaultProps = { ...Text.defaultProps, allowFontScaling: false };
```

File: src/index.js [hunk 3]
```diff
@@ -38,9 +38,14 @@ const styles = StyleSheet.create({
...
 // $FlowFixMe
-Text.defaultProps.allowFontScaling = false;
+TextInput.defaultProps = {
+  // $FlowFixMe
+  ...(TextInput.defaultProps || {}),
```
Comment (gre) [inline src/index.js]:
```suggestion
  ...TextInput.defaultProps,
```

**Comment by gre** [comment]
need more test on all phones

**Comment by juan-cortes** [comment]
we might revisit this, it's an eye sore seeing it unmerged :trollface:
